gpflow/base.py

```diff
@@ -112,6 +113,13 @@ def trainable(self, flag: Union[bool, int]):
     def initial_value(self):
         return self._unconstrained.initial_value
 
+    def validate_unconstrained_value(self, value: tf.Tensor, dtype: DType) -> tf.Tensor:
+        value = _cast_to_dtype(value, dtype)
+        unconstrained_value = _to_unconstrained(value, self.transform)
+        return tf.debugging.assert_all_finite(unconstrained_value,
+                message=f"gpflow.Parameter.validate_unconstrained_value")
```

Comment:
Could you change the message to something more descriptive?

Reply:
What would you suggest?

Reply:
"Unconstrained value of passed value has `NaN` or `Inf` and cannot be assigned"